driver/src/test/java/oracle/nosql/driver/ProxyTestBase.java

```diff
@@ -542,4 +576,105 @@ protected static void verbose(String msg) {
             System.out.println(msg);
         }
     }
+
+    /*
+     * convert a version string in X.Y.Z format to an
+     * integer value of (X * 1M) + (Y * 1K) + Z
+     * return -1 if the string isn't in valid X.Y.Z format
+     */
+    protected static int getIntegerVersion(String version) {
+        if (version == null || version.length() < 5) {
+            return -1;
+        }
+        String[] arr = version.split("\\.");
+        if (arr == null || arr.length != 3) {
+            return -1;
+        }
+        try {
+            return (Integer.parseInt(arr[0]) * 1000000) +
+                   (Integer.parseInt(arr[1]) * 1000) +
+                   Integer.parseInt(arr[2]);
+        } catch (Exception e) {}
+        return -1;
+    }
+
+    private static String getKVVersion(NoSQLHandle handle) {
+        /*
+         * Currently the proxy will only return the KV version that it was
+         * built against, and not the actual kv version of the store it's
+         * connected to.
+         * To enable testing with different proxy/kv combinations, allow
+         * an environment setting to override the kv value returned from
+         * the proxy.
+         */
+        String kvver = System.getenv("KV_VERSION");
```

Comment:
How does this work when the feature being tested relies on the kv client version vs the server version?

Reply:
That's a good question. Right now the assumption is that the kv client/server versions are the same. Maybe in the future we add a third version value?

Reply:
They can definitely differ which is why I didn't attempt to send back the server version from the proxy as it can vary with each request, especially in the cloud. I was thinking that your environment variable would set just the kv server version since the kv client version is reliable for a given proxy. That allows the 3 versions